unix/x0vncserver/x0vncserver.cxx

```diff
@@ -334,12 +336,15 @@ int main(int argc, char** argv)
     exit(1);
   }
 
+  const char *dname = XDisplayName(displayname);
   if (!(dpy = XOpenDisplay(displayname))) {
     // FIXME: Why not vlog.error(...)?
     fprintf(stderr,"%s: Unable to open display \"%s\"\r\n",
-            programName, XDisplayName(displayname));
+            programName, dname);
     exit(1);
   }
+  strncpy(displayStr, dname, 31);
+  displayStr[31] = '\0';
```

Comment:
We've generally switched to using `std::string` to avoid risky code like this. Hopefully we can do the same here.